app.js

```diff
@@ -15,12 +27,61 @@ app.use(express.static(__dirname + '/public'));
 // setting the spotify-api goes here:
 
 
-
-
+spotifyApi.clientCredentialsGrant()
+  .then( data => {
+    spotifyApi.setAccessToken(data.body['access_token']);
+  })
+  .catch(error => {
+    console.log('Something went wrong when retrieving an access token', error);
+  })
 
 
 // the routes go here:
 
+app.get('/', (req, res) => {
+  res.render('artists');
+});
+
+app.get(`/artists`, (req,res,) => {
+  console.log(req.query)
+  spotifyApi.searchArtists(req.query.text)
+  .then(data => {
+    console.log(data.body.artists.items[0]);
+    console.log(data.body.artists.items[0].id);
+    //console.log("The received data from the API: ", data.body);
+    res.render(`artistsFile`, data.body.artists)
```

Comment:
Maybe you could create an object before passing it to the view. It is more readable. Like `const artists = data.body`